src/expr/src/expr/build.rs

```diff
@@ -42,15 +42,24 @@ use crate::{bail, ExprError, Result};
 pub fn build_from_prost(prost: &ExprNode) -> Result<BoxedExpression> {
     use PbType as E;
 
-    match prost.expr_type() {
+    let func_call = match prost.get_rex_node()? {
+        RexNode::InputRef(_) => return InputRefExpression::try_from_boxed(prost),
+        RexNode::Constant(_) => return LiteralExpression::try_from_boxed(prost),
+        RexNode::Udf(_) => return UdfExpression::try_from_boxed(prost),
+        RexNode::FuncCall(func_call) => func_call,
+    };
+
+    let func_type = prost.function_type();
+
+    match func_type {
+        E::Unspecified => unreachable!(),
```

Comment:
Maybe same as `_` below to keep the old behavior of reporting an error for this request rather than panicking. This case may include unrecognized enum values beyond those manually set for input / value / udf.